### ISSUE TITLE
Base Score

### ISSUE THREAD

**Issue by Philipp-Sc** [zemlyansky/importance#1]
Hi Anton,

thank you for making this implementation public. It is the only JavaScript implementation I could find.
I am using it in my project. I have one question.

Why does the base score use the labels from the ground truth and not predictions calculated from the model?
Does this matter or is this a matter of choice?

Best regards,
Philipp

**Comment by zemlyansky** [comment]
Hi Philipp, thanks for your kind words. It's nice to hear that you are using the package. The main idea of the permutation importance is to measure how a model score (e.g., MAE, MSE, accuracy, etc.) changes when features are permuted. That's why we need those true labels to calculate a model score comparing true values with predicted ones.

**Comment by Philipp-Sc** [comment]
Thank you for your answer. This makes sense. The model score is measuring the accuracy of the model against the training data. (How well the model explains the data it was trained with).

So the base score is how good the models predicts the labels of the given input data without permutations.

I guess if the model is very accurate, one can just use the models predictions, with the goal to find out how sensitive the model reacts to changes when the features are permuted.
In that case the base score would be "perfect", i.e would not be needed to be calculated at all.

In this case we have no information how good the model is, but how much change the permutations cause which is a slightly different use case.

**Comment by zemlyansky** [comment]
Yeah, that's a good point. Thanks! 
What's interesting is permutation importance evaluated on true labels in some rare cases gives negatives values. That means that the model performed better with a permuted feature. Maybe it points to some pathology in data, a model or just appears because of randomness ¯\_(ツ)_/¯

**Comment by Philipp-Sc** [comment]
Thanks for helping me understand this. Yeah, I was wondering why in some cases I get negative numbers, though they are really small in my case, might as well just round up to zero.

Thanks again. I close this issue now, my questions are answered :+1:
